Mage.Tests/src/test/java/org/mage/test/cards/single/mh3/AuroraShifterTest.java

```diff
@@ -0,0 +1,60 @@
+package org.mage.test.cards.single.mh3;
+
+import mage.constants.PhaseStep;
+import mage.constants.Zone;
+import mage.counters.CounterType;
+import mage.players.Player;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mage.test.serverside.base.CardTestPlayerBase;
+
+public class AuroraShifterTest extends CardTestPlayerBase {
+
+    /**
+     * {@link mage.cards.a.AuroraShifter Aurora Shifter} {1}{U}
+     * Creature - Shapeshifter
+     * Changeling
+     * Whenever Aurora Shifter deals combat damage to a player, you get that many {E}.
+     * At the beginning of combat on your turn, you may pay {E}{E}.
+     * When you do, Aurora Shifter becomes a copy of another target creature you control,
+     * except it has this ability and "Whenever this creature deals combat damage to a player, you get that many {E}."
+     *
+     * 1/3
+     */
+    private static final String shifter = "Aurora Shifter";
+
+    private static void checkEnergyCount(String message, Player player, int expected) {
+        Assert.assertEquals(message, expected, player.getCountersCount(CounterType.ENERGY));
+    }
+
+    @Test
+    public void test_4_combats() {
+        setStrictChooseMode(true);
+
+        addCard(Zone.BATTLEFIELD, playerA, shifter);
+        addCard(Zone.BATTLEFIELD, playerA, "Grizzly Bears");
+
+//        Combat 1
+        attack(1, playerA, shifter, playerB);
+        runCode("energy counter is 1", 1, PhaseStep.POSTCOMBAT_MAIN, playerA, (info, player, game) -> checkEnergyCount(info, player, 1));
+        setStopAt(1, PhaseStep.END_TURN);
+        execute();
```

Comment:
don't call execute more than once in a test - only at the end, use checkXXX if you need interim asserts